File: jiho/캐시.cpp
#include <string>
#include <algorithm>
#include <vector>

using namespace std;

int solution(int cacheSize, vector<string> cities) 
{
    int answer = 0;
    
    vector<string>cache;
    
    for(string city : cities)
    {
        transform(city.begin(), city.end(), city.begin(), ::tolower);
        
        auto it = find(cache.begin(), cache.end(), city);
        if(it != cache.end())
        {
            cache.erase(it);
            cache.push_back(city);
            answer++;
        }
        else
        {
            if(cache.size() == cacheSize && !cache.empty())
                cache.erase(cache.begin());
            if(cache.size() < cacheSize)
                cache.push_back(city);
            answer += 5;   
        }
    }
    return answer;
}

File: yujin/캐시.py
from collections import deque

def solution(cacheSize, cities):
    answer = 0
    cache = deque(maxlen=cacheSize)
    
    for city_str in cities:
        city = city_str.lower()
        
        if cacheSize != 0 and city in cache:
            cache.remove(city)
            cache.append(city)
            answer += 1
        else:
            cache.append(city)
            answer += 5
            
    return answer
